control/eiger/eiger_detector.py

```diff
@@ -682,15 +690,25 @@ def initialize_detector(self):
         logging.info("Initializing the detector")
         self._initialize_event.set()
 
-    def fetch_stale_parameters(self):
-        for cfg in self._stale_parameters:
+    def fetch_stale_parameters(self, blocking=False):
+        with self._lock:
+            stale_parameters = set(self._stale_parameters)
+            self._stale_parameters.clear()
+
+        for cfg in stale_parameters:
             if cfg in self.DETECTOR_CONFIG:
                 param = self.read_detector_config(cfg)
             elif cfg in self.STREAM_CONFIG:
                 param = self.read_stream_config(cfg)
             logging.info("Read from detector [{}]: {}".format(cfg, param))
             setattr(self, cfg, param)
-        self._stale_parameters = []
+
+        # _stale_parameters may have been repopulated again by the time we get here and
+        # the flag will stay set - if so we fetch again here or in the next status poll
+        if blocking:
+            logging.info("Stale parameters repopulated - fetching again")
+            self.fetch_stale_parameters(blocking)
+
```

Comment:
Have I misunderstood this piece of code.  If blocking=True wouldn't you end up in an endless recursive loop?

Reply:
Only if puts are continuously sent that mean things need to be fetched.

Reply:
Sorry you are right. How about now?

Reply:
OK then.